keras_cv/models/segmentation/fcn.py

```diff
@@ -0,0 +1,677 @@
+        input_shape: `list` or `tuple`. Defines the shape of the tensor to be expected as input.
+        include_rescaling: bool, one of True or False. Defines whether to use a Rescaling layer or not.
+        return_mask: bool, one of True or False. Returns a 1-channel result instead of a
+            num_classes-channel result.
+        return_dtype: `tf.dtypes`. Return results in the chosen dtype. Defaults to tf.float32
+        include_imagenet_weights: bool, one of True or False. Defines whether to use
+            original ImageNet weights for VGG backbones.
+    """
+
+    def __init__(
+        self,
+        classes,
+        backbone,
+        model_architecture=None,
+        input_shape=(224, 224, 3),
+        include_rescaling=False,
+        return_mask=False,
+        return_dtype=tf.float32,
+        include_imagenet_weights=False,
+    ):
+
+        if isinstance(backbone, tf.keras.models.Model):
+            if model_architecture is not None:
+                raise ValueError(
+                    "`model_architecture` cannot be set if `backbone` is not in ['vgg16', 'vgg19']. Either set `backbone` to one of the accepted values or remove the `model_architecture` argument."
+                )
+            if include_imagenet_weights:
+                raise ValueError(
+                    """`include_imagenet_weights` cannot be set if `backbone` is not in ['vgg16', 'vgg19']. Either set `backbone` to one of the accepted values or remove the
+                    `include_imagenet_weights` argument."""
+                )
+            else:
+                backbone = CustomArchitectureBuilder(backbone)
+                classes_conv = tf.keras.layers.Conv2D(
+                    filters=classes,
+                    kernel_size=(1, 1),
+                    strides=(1, 1),
+                    padding="same",
+                    activation="softmax",
+                )
+
+                input_tensor = tf.keras.Input(input_shape)
+                if include_rescaling:
+                    x = Rescaling(scale=1.0 / 255)(input_tensor)
+                    x = backbone(x)
+                else:
+                    x = backbone(input_tensor)
+
+                output_shape = x.shape
+
+                target_height_factor = input_shape[0] // output_shape[1]
+                target_width_factor = input_shape[1] // output_shape[2]
+
+                upscale = tf.keras.layers.UpSampling2D(
+                    size=(target_height_factor, target_width_factor),
+                    data_format="channels_last",
+                    interpolation="bilinear",
+                )
+
+                x = classes_conv(x)
+                output_tensor = upscale(x)
+                if return_mask:
+                    # Assumes channels_last
+                    output_tensor = tf.math.argmax(output_tensor, axis=3)
+                    output_tensor = tf.expand_dims(output_tensor, axis=3)
+
+                if return_dtype != output_tensor.dtype:
```

Comment:
We should't expose this as an argument.
If we have to force the output dtype for stability, we should

Reply:
Alright, I'll make the change such that it casts it at all times to `tf.float32`